### ISSUE TITLE
Error when toString()'ing achievement messages

### ISSUE THREAD

**Issue by 1b8** [andrewrk/mineflayer#339]
It comes up with an error on line 181 of lib\chat_message.js:
```
TypeError: Cannot read property 'forEach' of undefined
```

**Comment by rom1504** [comment]
When do you get that error exactly ?

**Comment by 1b8** [comment]
Right after someone on the server earns an achievement (or gets one taken away by the `/achievement take` command), and the `.toString()` method is called on the message.

**Comment by rom1504** [comment]
I just tried with the chatterbox.js example:
* /achievement give achievement.overkill <bot_name>
* /achievement take achievement.overkill <bot_name>
* /achievement give achievement.overkill <an_other_player_name>
* /achievement take achievement.overkill <an_other_player_name>

None of this trigger any crash.

Can you provide an example triggering that error ?

**Comment by 1b8** [comment]
It crashes when calling the `toString()` method of the message. Example:

```javascript
var mineflayer = require('mineflayer');

var bot = mineflayer.createBot({
  // ...
});

bot.on('message', function (msg) {
  console.log(msg.toString());
});
```

then give/take an achievement:

```javascript
blah\mineflayer\lib\chat_message.js:181
    this.with.forEach(function(entry) {
             ^

TypeError: Cannot read property 'forEach' of undefined
```

**Comment by 1b8** [comment]
The ChatMessage for `1b8 has just earned the achievement [Taking Inventory]` looks like:
```json
{
  "json": {
    "translate": "chat.type.achievement",
    "with": [
      {
        "insertion": "1b8",
        "clickEvent": {
          "action": "suggest_command",
          "value": "/msg 1b8 "
        },
        "hoverEvent": {
          "action": "show_entity",
          "value": "{id:\"34eb9e87-564d-44a1-8591-8608ab491376\",name:\"1b8\"}"
        },
        "text": "1b8"
      },
      {
        "color": "green",
        "hoverEvent": {
          "action": "show_achievement",
          "value": "achievement.openInventory"
        },
        "extra": [
          {
            "color": "green",
            "hoverEvent": {
              "action": "show_achievement",
              "value": "achievement.openInventory"
            },
            "translate": "achievement.openInventory"
          },
          "]"
        ],
        "text": "["
      }
    ]
  },
  "translate": "chat.type.achievement",
  "with": [
    {
      "json": {
        "insertion": "1b8",
        "clickEvent": {
          "action": "suggest_command",
          "value": "/msg 1b8 "
        },
        "hoverEvent": {
          "action": "show_entity",
          "value": "{id:\"34eb9e87-564d-44a1-8591-8608ab491376\",name:\"1b8\"}"
        },
        "text": "1b8"
      },
      "text": "1b8",
      "clickEvent": {
        "action": "suggest_command",
        "value": "/msg 1b8 "
      },
      "hoverEvent": {
        "action": "show_entity",
        "value": "{id:\"34eb9e87-564d-44a1-8591-8608ab491376\",name:\"1b8\"}"
      }
    },
    {
      "json": {
        "color": "green",
        "hoverEvent": {
          "action": "show_achievement",
          "value": "achievement.openInventory"
        },
        "extra": [
          {
            "color": "green",
            "hoverEvent": {
              "action": "show_achievement",
              "value": "achievement.openInventory"
            },
            "translate": "achievement.openInventory"
          },
          "]"
        ],
        "text": "["
      },
      "text": "[",
      "extra": [
        {
          "json": {
            "color": "green",
            "hoverEvent": {
              "action": "show_achievement",
              "value": "achievement.openInventory"
            },
            "translate": "achievement.openInventory"
          },
          "translate": "achievement.openInventory",
          "color": "green",
          "hoverEvent": {
            "action": "show_achievement",
            "value": "achievement.openInventory"
          }
        },
        "]"
      ],
      "color": "green",
      "hoverEvent": {
        "action": "show_achievement",
        "value": "achievement.openInventory"
      }
    }
  ]
}
```

**Comment by 1b8** [comment]
What it seems to do is (`msg` being the message):
Loops through `msg.with`:
  `0`: Adds `"1b8 "` to `message`
  `1`: Adds `"[ "` to `message`
         Loops through `extra`:
           `0`: Looks for `text`, doesn't find.
                  Tries to loop through `with`, but throws error because it doesn't exist.
So the error is because `msg.with[1].extra[0]` isn't a string and does not have `text` or `with` fields.

**Comment by rom1504** [comment]
I just checked if with exists, so of course it doesn't provide an useful message for achievements like the vanilla client does, we might want to do this but we need https://github.com/PrismarineJS/minecraft-data/issues/17 first.

**Comment by rom1504** [comment]
@1b8 released in 1.5.0
